test/extensions/resource_monitors/injected_resource/config_test.cc

```diff
@@ -0,0 +1,38 @@
+#include "envoy/config/resource_monitor/injected_resource/v2alpha/injected_resource.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "common/event/dispatcher_impl.h"
+
+#include "server/resource_monitor_config_impl.h"
+
+#include "extensions/resource_monitors/injected_resource/config.h"
+
+#include "test/test_common/environment.h"
+#include "test/test_common/test_time.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ResourceMonitors {
+namespace InjectedResourceMonitor {
+
+TEST(InjectedResourceMonitorFactoryTest, CreateMonitor) {
+  auto factory =
+      Registry::FactoryRegistry<Server::Configuration::ResourceMonitorFactory>::getFactory(
+          "envoy.resource_monitors.injected_resource");
+  EXPECT_NE(factory, nullptr);
```

Comment:
You probably want to use `ASSERT_NE` here, since we can't safely proceed if it returned nullptr.